include/sound/sof/stream.h

```diff
@@ -83,10 +83,10 @@ struct sof_ipc_stream_params {
 	uint16_t sample_valid_bytes;
 	uint16_t sample_container_bytes;
 
-	/* for notifying host period has completed - 0 means no period IRQ */
 	uint32_t host_period_bytes;
+	uint32_t irq_mode; /* 1 means IRQ mode ON */
```

Comment:
you've wasting an entire 32-bit field for a boolean. Maybe use 16 bits and add another 16-bit as reserved?

Reply:
@plbossart, yes good suggestion. I have just updated.